docs/blog/2023-03-28-timelock-on-fastnet.md

```diff
@@ -0,0 +1,150 @@
+
+In case you were worried after reading our paper, know that whenever new members join the League, we conduct a so-called “resharing” of the League’s group secret key using verifiable secret sharing. This means that **the public key of the drand networks we are running never changes** and it is not going to cause an incompatibility issue with our timelock scheme. 
+
+### Our “timed release encryption” scheme
+
+Our timelock scheme, also sometimes called a “timed release encryption” (TRE) scheme, is a direct application of the [Identity Based Encryption scheme](https://crypto.stanford.edu/~dabo/papers/bfibe.pdf) (IBE) from Boneh et al. (section 4.2) to the threshold BLS signature setting of drand.  
+
+In IBE, participants are referred to via their **identity** which can be an email, a name, etc. and anyone can encrypt a message to any identity, even if the recipient is not “registered” with the system. This works because there is a Private Key Generator (PKG) service that distributes the required secret keys to participants depending on their identity.
+
+As it turns out, the League of Entropy is basically acting as a **“Distributed PKG” network** because they’re using the BLS scheme to sign messages and, (as already denoted in the initial 2001 IBE paper), the decryption keys of their IBE scheme can be used as signatures over a given identity, which is exactly what the BLS scheme is accomplishing. 
+
+Moreover, it uses **the round number as the identity**, **and the corresponding signature becomes the private key to decrypt.** In short:
+
+Public key = **round number**
+
+Private key = **signature over the round number**
+
+Given this, the flow is as follows:
+
+- Anyone can encrypt a message towards a specific round X in advance
+- When the time has come, the network will generate the BLS signature (i.e. the *private key)* and release it publicly
+- Anybody can retrieve the BLS signature of the beacon produced at round X and decrypt any message that was encrypted towards that round X.
+
+For more technical information about how this works, you can **check our pre-print paper** about our Timelock Encryption scheme on ePrint [here](https://eprint.iacr.org/2023/189).
+
+## Hybrid encryption
+
+Because our timed release encryption scheme allows users to **encrypt a fixed sized message**, and also to achieve better performance in general, it relies on “**Hybrid encryption**” to encrypt arbitrary data more easily. Hybrid encryption means that we are using timed encryption to **encrypt a Data Encryption Key (DEK)** used to subsequently encrypt the actual data we want to transmit with “regular” encryption schemes. This is commonly known as “wrapping a key”, and its how modern public key encryption schemes function to encrypt more data than their “block size”. Typically, the DEK is an AES or ChaCha key, both being blazingly fast symmetric encryption schemes compared to public key scheme used to encrypt the DEK.
```

Comment:
```suggestion
Because our timed release encryption scheme allows users to **encrypt a fixed sized message**, and also to achieve better performance in general, it relies on “**Hybrid encryption**” to encrypt arbitrary data more easily. Hybrid encryption means that we are using timed encryption to **encrypt a Data Encryption Key (DEK)** used to subsequently encrypt the actual data we want to transmit with “regular” encryption schemes. This is commonly known as “wrapping a key”, and it's how modern public key encryption schemes function to encrypt more data than their “block size”. Typically, the DEK is an AES or ChaCha key, both being blazingly fast symmetric encryption schemes compared to the public key scheme used to encrypt the DEK.
```